common/dds_specific/srcCxx/DynamicDataHelpers.cxx

```diff
@@ -385,3 +385,172 @@ DynamicType rti::common::dynamic_data::get_member_type(
     }
     return *type;
 }
+
+
+/*
+ * @brief Copy a primitive member from the input to the output. The index
+ * specifies which type is being copied.
+ *
+ * Input may be a sequence or an array
+ * Output is an array
+ */
+void rti::common::dynamic_data::copy_primitive_array_elements(
```

Comment:
the `DynamicData::set_values()` and `DynamicData::get_values()` work for sequences and arrays, however, as we are assigning always the `max_elements` as the output size (fixed), this should only be used with arrays.
With sequences, we don't need to do that because the `values` array will have the current (and correct) size.

Probably I should enforce that the input may be an array or sequence, but the output **must** be an array.